### PR TITLE
Persist pm.variables across scripts

### PR DESCRIPTION
**Pull request by harryi3t**
~Blocked on https://github.com/postmanlabs/postman-sandbox/pull/223~

### CODE REVIEW

**Comment by harryi3t** [comment]
@kamalaknn Tests are failing coz sandbox can't be used from any github branch.
Since it needs the `.cache/bootcode` to be present which is ignored in git but not in npm.
We need a released version for the tests to pass.

File: lib/runner/extensions/event.command.js [hunk 1]
```diff
@@ -328,9 +329,16 @@ module.exports = {
                     // @todo - should this be handled by the sandbox?
                     result && result.environment && (result.environment = new sdk.VariableScope(result.environment));
                     result && result.globals && (result.globals = new sdk.VariableScope(result.globals));
+                    result && result.collectionVariables &&
+                        (result.collectionVariables = new sdk.VariableScope(result.collectionVariables));
```
Comment (kamalaknn) [inline lib/runner/extensions/event.command.js]: @shamasis we are bubbling `_variables` up with the results on the `test` callback. I'm guessing we are okay with this?

File: lib/runner/request-helpers-presend.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ module.exports = [
         // @todo - no need to sync vatiables when SDK starts supporting resolution from scope directly
         // @todo remove this hack of using empty variableList once it's fixed in sdk
         context.item = new sdk.Item(context.item.toObjectResolved({variables: new sdk.VariableList(null, [])}, [
+            context._variables.syncVariablesTo({}),
```
Comment (kamalaknn) [inline lib/runner/request-helpers-presend.js]: @shamasis highlighting this change, `pm.variables.set` takes the highest precedence, higher than iteration data, the same as in `sandbox`

File: test/integration/inherited-entities/pm-variables.test.js [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+        before(function (done) {
+            var clonedRunOptions = _.merge({}, runOptions, {
+                collection: {
+                    event: [
+                        {
+                            listen: 'prerequest',
+                            script: {
+                                exec: `
+                                // only run for 1st item
+                                if (pm.variables.get('key-1') !== 'modified-1') {
```
Comment (kamalaknn) [inline test/integration/inherited-entities/pm-variables.test.js]: @harryi3t can we try to use `unset` and avoid logic in the test fixtures to make it easy to comprehend?

Comment (harryi3t) [reply test/integration/inherited-entities/pm-variables.test.js]: ok will try it out

File: lib/runner/request-helpers-presend.js [hunk 4]
```diff
@@ -14,9 +14,10 @@ module.exports = [
         if (!context.item) { return done(new Error('Nothing to resolve variables for.')); }
 
         // @todo - resolve variables in a more graceful way
+
         // @todo - no need to sync variables when SDK starts supporting resolution from scope directly
-        context.item = new sdk.Item(context.item.toObjectResolved(null, [context.data,
-            context.environment.syncVariablesTo({}), context.collectionVariables.syncVariablesTo({}),
+        context.item = new sdk.Item(context.item.toObjectResolved(null, [context._variables.syncVariablesTo({}),
```
Comment (shamasis) [inline lib/runner/request-helpers-presend.js]: @kamalaknn - wondering if we should make this resolution as part of utils? It is used in two primary places?

```
resolveProperty = function (prop, context) ...
```

**Pull request merged**